src/styles/style-helpers.js

```diff
@@ -61,3 +61,35 @@ export const disableVisually = () => `
   opacity: 0.5;
   pointer-events: none;
 `;
+
+const add = (a, b) => a + b;
+const subtract = (a, b) => a - b;
+const multiply = (a, b) => a * b;
+const divide = (a, b) => a / b;
+
+const transformUnit = (values, transform, allowMultipleUnits = true) => {
+  const checkUnit = (aUnit, bUnit) => {
+    if (!allowMultipleUnits && aUnit && bUnit) {
+      throw Error(`You cannot ${transform.name} multiple values with a unit.`);
+    }
+    if (aUnit && bUnit && aUnit !== bUnit) {
+      throw Error(`You cannot ${transform.name} values with different units.`);
+    }
+  };
+  const transformedValue = values.reduce((result, value) => {
```

Comment:
I think your function will break if you do something like `transformUnit(['2px', 3, '3px'], multiply, false)`. At least it won't behave as expected.

Reply:
In this example, it throws an error as it should since you're trying to multiply two pixel values (which logically doesn't work and you never really want to do anyway).